pkg/proc/evalop/evalcompile.go

```diff
@@ -483,25 +483,32 @@ func (ctx *compileCtx) compileReslice(node *ast.SliceExpr) error {
 		return err
 	}
 
+	trustLen := true
 	hasHigh := false
 	if node.High != nil {
 		hasHigh = true
 		err = ctx.compileAST(node.High)
 		if err != nil {
 			return err
 		}
+		_, isbasiclit := node.High.(*ast.BasicLit)
+		trustLen = trustLen && isbasiclit
+	} else {
+		trustLen = false
 	}
 
 	if node.Low != nil {
 		err = ctx.compileAST(node.Low)
 		if err != nil {
 			return err
 		}
+		_, isbasiclit := node.Low.(*ast.BasicLit)
```

Comment:
Why perform this check on the low side, what would cause us to not trust the length depending on the low side of the slice operation?

Reply:
My thinking is that we only do this when the expression has a fixed length and something like `s[a:1000000]` doesn't.